client/src/components/MultiStep/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Box } from '@material-ui/core';
+import BasicUserInfo from '../signUpForm/BasicUserInfo';
+import GenderSelection from '../signUpForm/GenderSelection';
+import PhysicalCharacteristics from '../signUpForm/PhysicalCharacteristics';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+  },
+  button: {
+    marginRight: theme.spacing(1),
+  },
+  instructions: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export default function HorizontalLinearStepper() {
+  const classes = useStyles();
+  const [activeStep, setActiveStep] = useState(0);
+  const [skipped, setSkipped] = useState(new Set());
+  const [data, setData] = useState({});
+
+  const isStepSkipped = (step) => skipped.has(step);
+
+  const handleNext = () => {
+    let newSkipped = skipped;
+    if (isStepSkipped(activeStep)) {
+      newSkipped = new Set(newSkipped.values());
+      newSkipped.delete(activeStep);
+    }
+
+    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    setSkipped(newSkipped);
+  };
+
+  const handleBack = () => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  };
+
+  useEffect(() => {
+    console.log(data);
+  }, [data]);
+
+  function getStepContent(step) {
```

Comment:
This is logic that could move into a subcomponent (in a new file) which takes `step`, `handleBack`, `handleNext` and `setData` props.

```
export default StepContent = ({step, handleBack, handleNext, setData}) => {
    switch (step) {
      case 0:
        return <BasicUserInfo methods={{ handleBack, handleNext, setData }} />;
      case 1:
        return (
          <GenderSelection methods={{ handleBack, handleNext, setData }} />
        );
      case 2:
        return (
          <PhysicalCharacteristics
            methods={{ handleBack, handleNext, setData }}
          />
        );
      default:
        return <BasicUserInfo methods={{ handleBack, handleNext, setData }} />;
    }
};
```
This would be better because then React can detect when the props change and will only re-render this component when there's a prop change.  WIth your current way, React will be re-rendering this content every time this HorizontalMultiStep component renders, which could be less efficient.